src/legacy/server/saved_objects/routes/export.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Joi from 'joi';
+import stringify from 'json-stable-stringify';
+import { getSortedObjectsForExport } from '../lib/export';
+
+const EXPORT_SIZE_LIMIT = 10000;
+
+export const createExportRoute = (prereqs: any) => ({
+  path: '/api/saved_objects/_export',
+  method: 'GET',
+  config: {
+    pre: [prereqs.getSavedObjectsClient],
+    validate: {
+      query: Joi.object()
+        .keys({
```

Comment:
Should limit types until further PR allows export of any type.

Reply:
When @njd5475 explored using the route schema validation to limit the types previously, it became an information disclosure issue because anonymous users can determine which plugins are registered via which types are valid here. We ended up relying on the SavedObjectsClient and the Repository to enforce the type restrictions in these scenarios.

Reply:
Sorry I should have added more details, I want to restrict only: `dashboard`, `visualization`, `index-pattern` & `search` for now to replicate current export functionality. Without those, users would be able to export any type which I haven't prepared for yet (future PR).